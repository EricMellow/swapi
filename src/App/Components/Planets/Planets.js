import React from 'react';
import './Planets.css';

const Planets = ({cardData}) => {
  let planetCards = [];
  if (cardData) {
    planetCards = cardData.map((planet, index) => {
      return (
        <div key={planet.name + index} className="card">
          <h3>{planet.name}</h3>
          <p>Terrain: {planet.terrain}</p>
          <p>Climate: {planet.climate}</p>
          <p>Population: {planet.population}</p>
          <p>Residents: {planet.residents}</p>
        </div>
      );
    });

  }

  return (
    <div className="container">
      {planetCards.length > 1 && planetCards}
    </div>
  );
};

export default Planets;