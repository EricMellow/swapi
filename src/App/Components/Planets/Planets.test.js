import React from 'react';
import Planets from "./Planets.js";
import { shallow } from "enzyme";
import { mockedCleanedPlanets } from '../../mockPlanetData';

describe('Planets', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Planets />);
  });

  it('should match the snapshot with no data', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with good data', () => {
    const mockData = mockedCleanedPlanets;
    wrapper = shallow(<Planets cardData={mockData} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with error data', () => {
    const mockErrorData = [{
      terrain: "Something went wrong",
      name: "Oh no!",
      population: "No data",
      climate: ":(",
      residents: "No data"
    }];
    wrapper = shallow(<Planets cardData={mockErrorData} />);

    expect(wrapper).toMatchSnapshot();
  });
});