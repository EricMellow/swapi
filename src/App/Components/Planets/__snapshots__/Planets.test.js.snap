// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planets should match the snapshot with error data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    cardData={
        Array [
            Object {
              "climate": ":(",
              "name": "Oh no!",
              "population": "No data",
              "residents": "No data",
              "terrain": "Something went wrong",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
      "className": "container",
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
        "className": "container",
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Planets should match the snapshot with good data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets
    cardData={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
      "className": "container",
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
        "className": "container",
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Planets should match the snapshot with no data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Planets />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": false,
      "className": "container",
    },
    "ref": null,
    "rendered": false,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": false,
        "className": "container",
      },
      "ref": null,
      "rendered": false,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
